core/src/main/java/org/apache/iceberg/expressions/ExpressionParser.java

```diff
@@ -51,6 +52,11 @@ public class ExpressionParser {
   private static final String REFERENCE = "reference";
   private static final String LITERAL = "literal";
 
+  public enum SerializationMode {
```

Comment:
I'd prefer a separate parser (like `ResolvedExpressionParser`) or parser method (like `toResolvedJson`) rather than exposing an enum like this.